tests/connection_state_test.c

```diff
@@ -915,3 +915,20 @@ AWS_TEST_CASE_FIXTURE(
     s_test_mqtt_connection_offline_publish_fn,
     s_clean_up_mqtt_server_fn,
     &test_data)
+
+/**
+ * TODO: Another race condition may cause the publish results differently due to our inconsistent retry police.
```

Comment:
A workaround is possible, but if we are going to make change in the near future (next week), I will wait until the change to make this test.